sys/include/event/timeout.h

```diff
@@ -107,6 +107,17 @@ void event_timeout_set(event_timeout_t *event_timeout, uint32_t timeout);
  */
 void event_timeout_clear(event_timeout_t *event_timeout);
 
+/**
+ * @brief   Check if a timeout event is currently active
+ *
+ * @param[in]   event_timeout   event_timout context object to use
+ * @return      true if the event is scheduled, false otherwise
+ */
+static inline bool event_timeout_is_set(const event_timeout_t *event_timeout)
+{
+    return ztimer_is_set(event_timeout->clock, &event_timeout->timer);
```

Comment:
Now I wonder if this shouldn't rather be
```suggestion
static inline bool event_timeout_is_scheduled(const event_timeout_t *event_timeout)
{
    return ztimer_is_set(event_timeout->clock, &event_timeout->timer) ||
           event_is_queued(event_timeout->queue, event_timeout->event);
```

Reply:
Depends on what your function is supposed to express. The name implies to me that you want to know if the timeout already expired and the event is about to happen. I ended up using `event_timeout_is_scheduled() || event_is_queued())`, but the use case of knowing whether the timeout expired is also valid to me.
Take a look at the documentation of `event_timeout_clear()` it also only considers the timer, so your current implementation fits well in the line.

Reply:
Yea but the use case is to check if the event is set to happen in the future, so just checking `ztimer_is_set()` is racy as there is still the period when the event is queued but not executed.

In fact, I couldn't think of a use case where one would be interested in just the timeout being set but not the event being queued.

Reply:
Do you want to add both `event_timeout_is_set()` and `event_timeout_is_scheduled()` to not make a wrong decision on the behavior?

Reply:
I can't really think of a case where `event_timeout_is_set()` would be useful with checking ztimer alone, but I can think of many cases where using it would be buggy.